Cryptomator/Onboarding/OnboardingCoordinator.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  OnboardingWelcomeCoordinator.swift
+//  Cryptomator
+//
+//  Created by Tobias Hagemann on 08.09.21.
+//  Copyright © 2021 Skymatic GmbH. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class OnboardingCoordinator: Coordinator {
+	var childCoordinators = [Coordinator]()
+	var navigationController: UINavigationController
+
+	init(navigationController: UINavigationController) {
+		self.navigationController = navigationController
+	}
+
+	func start() {
+		let onboardingViewController = OnboardingViewController(viewModel: OnboardingViewModel())
+		onboardingViewController.coordinator = self
+		navigationController.pushViewController(onboardingViewController, animated: false)
+	}
+
+	func showIAP() {
+		let child = PurchaseCoordinator(navigationController: navigationController)
+		childCoordinators.append(child) // TODO: remove missing?
```

Comment:
We should definitely add child coordinators as this is the only strong reference to the coordinator.
However, you are right that the removal is not handled (correctly) at the moment.
But I think this can be solved in a generic way, fixing this for all the coordinators in the project.
Therefore, I would suggest that we fix that for the whole project after this PR.